e2e/steps/match.ts

```diff
@@ -106,5 +128,22 @@ export const matchAndBookApplication = async ({
     'Matched',
   ])
 
-  return { premisesName, premisesId }
+  return { premisesName, premisesId, newDatesOfPlacement }
```

Comment:
sending these `newDatesOfPlacement` to the confirmation page and confirming has a rippling effect on where we see thise new dates in the UI. Therefore returning here so that they can be tested in those places.